tokio/src/net/windows/named_pipe.rs

```diff
@@ -0,0 +1,1166 @@
+        bool_flag!(self.open_mode, allowed, winbase::PIPE_ACCESS_INBOUND);
+        self
+    }
+
+    /// The flow of data in the pipe goes from server to client only.
+    ///
+    /// This corresponds to setting [`PIPE_ACCESS_OUTBOUND`].
+    ///
+    /// [`PIPE_ACCESS_OUTBOUND`]: https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createnamedpipea#pipe_access_outbound
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::io;
+    /// use tokio::io::{AsyncReadExt as _, AsyncWriteExt as _};
+    /// use tokio::net::windows::named_pipe::{ClientOptions, ServerOptions};
+    ///
+    /// const PIPE_NAME: &str = r"\\.\pipe\tokio-named-pipe-access-outbound";
+    ///
+    /// # #[tokio::main] async fn main() -> io::Result<()> {
+    /// // Server side prevents connecting by denying outbound access, client errors
+    /// // when attempting to create the connection.
+    /// {
+    ///     let _server = ServerOptions::new()
+    ///         .access_outbound(false)
+    ///         .create(PIPE_NAME)?;
+    ///
+    ///     let e = ClientOptions::new()
+    ///         .open(PIPE_NAME)
+    ///         .unwrap_err();
+    ///
+    ///     assert_eq!(e.kind(), io::ErrorKind::PermissionDenied);
+    ///
+    ///     // Disabling reading allows a client to connect, but leads to runtime
+    ///     // error if a read is attempted.
+    ///     let mut client = ClientOptions::new()
+    ///         .read(false)
+    ///         .open(PIPE_NAME)?;
+    ///
+    ///     let mut buf = [0u8; 4];
+    ///     let e = client.read(&mut buf).await.unwrap_err();
+    ///     assert_eq!(e.kind(), io::ErrorKind::PermissionDenied);
+    /// }
+    ///
+    /// // A functional, unidirectional client-to-server only communication.
+    /// {
+    ///     let mut server = ServerOptions::new().access_outbound(false).create(PIPE_NAME)?;
+    ///     let mut client = ClientOptions::new().read(false).open(PIPE_NAME)?;
+    ///
+    ///     // TODO: Explain why this test doesn't work without calling connect
+    ///     // first.
+    ///     //
+    ///     // Because I have no idea -- udoprog
+    ///     server.connect().await?;
+    ///
+    ///     let write = client.write_all(b"ping");
+    ///
+    ///     let mut buf = [0u8; 4];
+    ///     let read = server.read_exact(&mut buf);
+    ///
+    ///     let ((), read) = tokio::try_join!(write, read)?;
+    ///
+    ///     println!("done reading and writing");
+    ///
+    ///     assert_eq!(read, 4);
+    ///     assert_eq!(&buf[..], b"ping");
+    /// }
+    /// # Ok(()) }
+    /// ```
+    pub fn access_outbound(&mut self, allowed: bool) -> &mut Self {
+        bool_flag!(self.open_mode, allowed, winbase::PIPE_ACCESS_OUTBOUND);
+        self
+    }
+
+    /// If you attempt to create multiple instances of a pipe with this flag,
+    /// creation of the first instance succeeds, but creation of the next
+    /// instance fails with [`ERROR_ACCESS_DENIED`].
```

Comment:
I think we should link to the `ErrorKind` for cases where you can check it via the error kind.

Reply:
done!